server.go

```diff
@@ -265,42 +313,55 @@ func (s *Server) getVectorTile(rctx context.Context, w io.Writer, r *http.Reques
 	fcLayers := make(mvt.Layers, len(layersToCompute))
 	for i, layer := range layersToCompute {
 		i, layer := i, layer // Fun stuff: https://blog.cloudflare.com/a-go-gotcha-when-closures-and-goroutines-collide/
+
+		// Start a goroutine for each layer
 		eg.Go(func() error {
-			Logger.Debugf("Retrieving vector tile for layer [%s] @ (%d, %d, %d)", layer.Name, x, y, z)
-			fc, err := layer.Source.GetFeatures(ctx, req)
+			Logger.Debugf("Retrieving layer data for [%s] @ (%d, %d, %d)", layer, z, x, y)
+
+			fcLayer, err := s.getLayerData(ctx, layer, req)
 			if err != nil {
 				return err
 			}
-			fcLayer := mvt.NewLayer(layer.Name, fc)
-			fcLayer.Version = 2 // Set to tile spec v2
-			fcLayer.ProjectToTile(req.MapTile())
-			fcLayer.Clip(mvt.MapboxGLDefaultExtentBound)
 
+			// TODO: Consider the tradeoffs of cacheing pre-simplified vs. post-simplified layers
 			if s.Simplify {
 				minZoom := layer.Minzoom
 				maxZoom := layer.Maxzoom
-				simplifyThreshold := calculateSimplificationThreshold(minZoom, maxZoom, z)
-				Logger.Debugf("Simplifying @ zoom [%d], epsilon [%f]", z, simplifyThreshold)
+				simplifyThreshold := calculateSimplificationThreshold(minZoom, maxZoom, req.Z)
```

Comment:
Any reason for `req.Z` vs `z`?

Reply:
Nope, this was unintentional. I'll fix it to make more consistent :+1: